### PR TITLE
Fix message "There are not enough products in stock" 

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | https://github.com/PrestaShop/PrestaShop/issues/12630
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/12630
| How to test?  | https://github.com/PrestaShop/PrestaShop/issues/12630

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12655)

### CODE REVIEW

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 1]
```diff
@@ -776,7 +776,7 @@ public function addQuantityInformation(
                     : Configuration::get('PS_LABEL_OOS_PRODUCTS_BOA', $language->id);
                 $this->product['availability_date'] = $product['available_date'];
                 $this->product['availability'] = 'available';
-            } elseif ($product['quantity_wanted'] > 0 && $product['quantity'] > 0) {
+            } elseif ($product['quantity_wanted'] > 1 && $product['quantity'] > 0) {
```
Comment (PierreRambaud) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]: Should be for a better understanding. 
```suggestion
            } elseif ($product['quantity'] - $product['quantity_wanted'] < 0) {
```

**Review by PierreRambaud — APPROVED** [review]
Let's try this!

**Comment by ntiepresta** [comment]
Hi,
This issue doesn't exist on develop branch.

Thanks

**Comment by MathiasReker** [comment]
Then I close this PR.

**Comment by khouloudbelguith** [comment]
Hi, 

Yes, I confirm that the issue doesn't exist in the develop branch.
@MathiasReker, you can close your PR.
Thanks!

**Comment by MathiasReker** [comment]
I already did :)

**Comment by Andrex777** [comment]
Hi a recently update my prestashop to the 1.7.6 version and i have the same issue. I try Pierre's code but it didnt work, how can i solve it? could you guys can help me?

**Comment by 8ternity** [comment]
This resolution [#12630](https://github.com/PrestaShop/PrestaShop/issues/12630) not worked for me. The default label are still showing the label "There are not enough products in stock".

I download a Fresh installation of 1.7.6.3 files and copy the line 788 of \src\Adapter\Presenter\Product\ProductLazyArray.php, replace the line 766 into the version of 1.7..5.1 currently in my store (same file). 
Replace
`} elseif ($product['quantity'] - $product['quantity_wanted'] < 0) {`
for
`} elseif ($product['quantity_wanted'] > 0 && $product['quantity'] > 0) {`

**And now it's fixed.** 
It was present in my version 1.7.5.1 and with the changes into the v1.7.6.3, it's working now with the label entered into "Product Settings".

**Pull request closed without merging**